### ISSUE TITLE
API doc？

### ISSUE THREAD

**Issue by linonetwo** [kezike/solid-vc#1]
Can I use this project to build some Web application that uses VC?

What's the JavaScript API? Is there a npm package?

**Comment by kezike** [comment]
Hi @linonetwo,

At the moment there is nothing of the sort. Part of my struggle while building this platform was thinking about the best way to expose this work to other ecosystems. On the one hand, I could either take the direction of developing a workflow that supports particular credential types. For example, imagine if selecting the credential type in the `Request` tab of the `Subject` interface prompted you to provide credential-specific information or upload identity proofs of some sort that would help the Issuer decide whether to give you a credential. That's one possible direction. The other possible direction that I think might be more constructive would be to allow an ecosystem of applications to develop around `solid-vc`, each specializing in a particular credential type. How exactly the wiring between `solid-vc` and these external applications would look has yet to be specified in any formal way and is open for discussion.

Another relevant API-like development that is in the works is a CLI that exposes the functionalities of this platform in a command-line environment. This will be a major development that I am looking forward to, as it would allow for automated services and would also be developer friendly :)

Kayode

**Comment by dmitrizagidulin** [comment]
@linonetwo @kezike So, myself and the Digital Bazaar team are currently working on several libraries related to this. For example:

https://github.com/digitalbazaar/jsonld-signatures - This library does the actual Linked Data Signatures that VCs use.

https://github.com/w3c/vc-test-suite - A test suite for the [VC spec](https://w3c.github.io/vc-data-model/)

https://github.com/digitalbazaar/vc-js/tree/dev a helper library to create and verify Verifiable Claims. It's currently undocumented, but we're actively working on it this week, to prepare it to pass the vc-test-suite :)

**Comment by kezike** [comment]
Yes, thank you @dmitrizagidulin for including this here! I meant to reference this library here, as it was used heavily in `solid-vc`. And @dmitrizagidulin and Digital Bazaar have been doing a great job in implementing tools for the linked data sigs and verifiable credentials specs in general!
